cmd/dm-ctl/main.go

```diff
@@ -31,37 +31,98 @@ import (
 	"github.com/pingcap/dm/pkg/utils"
 )
 
+func helpUsage(cfg *common.Config) {
+	fmt.Println("Usage: dmctl [global options] command [command options] [arguments...]")
+	fmt.Println()
+	ctl.Usage()
+	fmt.Println()
+	fmt.Println("Special Command:")
+	f := cfg.FlagSet.Lookup(common.EncryptCmdName)
+	fmt.Println(fmt.Sprintf("\t--%s %s", f.Name, f.Usage))
+	fmt.Println()
+	fmt.Println("Global Options:")
+	cfg.FlagSet.VisitAll(func(flag2 *flag.Flag) {
+		if flag2.Name == common.EncryptCmdName {
+			return
+		}
+		fmt.Println(fmt.Sprintf("\t--%s %s", flag2.Name, flag2.Usage))
+	})
+}
+
 func main() {
 	cfg := common.NewConfig()
-	err := cfg.Parse(os.Args[1:])
-	switch errors.Cause(err) {
-	case nil:
-	case flag.ErrHelp:
-		os.Exit(0)
-	default:
-		fmt.Printf("parse cmd flags err: %s", err)
-		os.Exit(2)
+	args := os.Args[1:]
+
+	// try to split one task operation from dmctl command
+	// because we allow user put task operation at last with two restrictions
+	// 1. one command one task operation
+	// 2. put task operation at last
+	cmdArgs := collectArgs(args)
+	lenArgs := len(args)
+	lenCmdArgs := len(cmdArgs)
+	if lenCmdArgs > 0 {
+		lenArgs = lenArgs - lenCmdArgs
 	}
 
-	utils.PrintInfo2("dmctl")
-	fmt.Println() // print a separater
+	if len(cmdArgs) > 0 {
+		commandMode(cmdArgs)
+	} else {
+		finished, err := cfg.Parse(args[:lenArgs])
+		if finished {
+			os.Exit(0)
+		}
 
-	// now, we use checker in dmctl while it using some pkg which log some thing when running
-	// to make dmctl output more clear, simply redirect log to file rather output to stdout
-	err = log.InitLogger(&log.Config{
-		File:  "dmctl.log",
-		Level: "info",
-	})
-	if err != nil {
-		fmt.Printf("init logger error %v", errors.ErrorStack(err))
-		os.Exit(2)
+		switch errors.Cause(err) {
+		case nil:
+		case flag.ErrHelp:
+			helpUsage(cfg)
+			os.Exit(0)
+		default:
+			fmt.Printf("parse cmd flags err: %s", err)
+			os.Exit(2)
+		}
+		// now, we use checker in dmctl while it using some pkg which log some thing when running
+		// to make dmctl output more clear, simply redirect log to file rather output to stdout
+		err = log.InitLogger(&log.Config{
+			File:  "dmctl.log",
+			Level: "info",
+		})
+		if err != nil {
+			fmt.Printf("init logger error %v", errors.ErrorStack(err))
+			os.Exit(2)
+		}
+
+		err = ctl.Init(cfg)
+		if err != nil {
+			fmt.Printf("init control error %v", errors.ErrorStack(err))
+			os.Exit(2)
+		}
+		interactionMode()
 	}
+}
 
-	err = ctl.Init(cfg)
-	if err != nil {
-		fmt.Printf("init control error %v", errors.ErrorStack(err))
-		os.Exit(2)
+func collectArgs(args []string) []string {
+	collectedArgs := make([]string, 0, len(args))
+	for i := 0; i < len(args); i++ {
+		if ctl.HasCommand(strings.ToLower(args[i])) {
```

Comment:
check multiple commands that exist or rely on `ctl.Start(args)` in L116? but may need to diff _command name_ with _command options_.